src/sentry/static/sentry/app/views/organizationDiscover/result/utils.jsx

```diff
@@ -231,3 +232,37 @@ const LightGray = styled.span`
 const DarkGray = styled.span`
   color: ${p => p.theme.gray5};
 `;
+
+/**
+ * Downloads a Snuba result object as CSV format
+ *
+ * @param {Object} result Result received from Snuba
+ * @param {Object} result.data Result data object from Snuba
+ * @param {String} result.meta Result metadata from Snuba
+ * @returns {Void}
+ */
+export function downloadAsCsv(result) {
+  const {meta, data} = result;
+  const headings = meta.map(({name}) => name);
+
+  const csvContent = Papa.unparse({
+    fields: headings,
+    data: data.map(row => {
+      return headings.map(col => quoteUnsafeStrings(row[col]));
+    }),
+  });
+
+  const encodedDataUrl = encodeURI(`data:text/csv;charset=utf8,${csvContent}`);
+
+  window.location.assign(encodedDataUrl);
+}
+
+function quoteUnsafeStrings(value) {
+  const unsafeCharacterRegex = /^[\=\+\-\@]/;
+
+  if (typeof value === 'string' && `${value}`.match(unsafeCharacterRegex)) {
+    return `\t"${value}"`;
```

Comment:
Can your data have `"` inside the fields? Right now `"` is not in your unsafe list and this quoting is not replacing the `"` with `""`.

Reply:
Ok yeah, in that case maybe I'd also add this to any field that starts with `"=`, `"+`, `=-`, `=@`

Reply:
Hmm, papa parse automatically escapes quotes though so maybe this is not required

Reply:
You can prevent macros by prepending a `'` to fields that start with a macro byte.